enterprise/frontend/src/embedding-sdk/cli/utils/nextjs-helpers.ts

```diff
@@ -0,0 +1,129 @@
+import fs from "fs";
+
+import path from "path";
+
+import { getNextJsCustomAppOrRootLayoutSnippet } from "../snippets/nextjs-app-snippets";
+import { getNextJsAnalyticsPageSnippet } from "../snippets/nextjs-page-snippet";
+
+import { checkIsInTypeScriptProject } from "./check-typescript-project";
+import { getProjectDependenciesFromPackageJson } from "./get-package-version";
+
+const hasPathInProject = (fileName: string) =>
+  fs.existsSync(path.join(process.cwd(), fileName));
+
+/**
+ * Checks if the current project is a Next.js project.
+ *
+ * It determines this by checking if the `tsconfig.json` file exists, or
+ * if the `package.json` file contains a `typescript` dev dependency.
+ */
+export async function checkIsInNextJsProject() {
+  const dependencies = await getProjectDependenciesFromPackageJson();
+
+  const hasNextJsDependency = !!dependencies?.next;
+
+  const hasNextJsConfig =
+    hasPathInProject("next.config.js") || hasPathInProject("next.config.ts");
+
+  return hasNextJsDependency || hasNextJsConfig;
+}
+
+/**
+ * Check if the current project is using the app or page router.
+ * Prioritizes the app router (more modern) if both are present.
+ */
+export async function checkIfUsingAppOrPagesRouter() {
+  if (hasPathInProject("app")) {
+    return "app";
+  }
+
+  if (hasPathInProject("pages")) {
+    return "pages";
+  }
+
+  return null;
+}
+
+/**
+ * Checks if the current Next.js project has a custom root layout (app router)
+ * or custom app (pages router).
+ *
+ * @see https://nextjs.org/docs/pages/building-your-application/routing/custom-app
+ * @see https://nextjs.org/docs/app/api-reference/file-conventions/layout#root-layouts
+ */
+export async function checkIfNextJsCustomAppOrRootLayoutExists() {
+  const router = await checkIfUsingAppOrPagesRouter();
+
+  // App router uses the `app/layout` root layout file.
+  if (router === "app") {
+    return (
+      hasPathInProject("app/layout.js") || hasPathInProject("app/layout.tsx")
```

Comment:
It's weird on their doc that it needs to either be `.js` or `.tsx` like wouldn't `.jsx` and `.ts` work too? I haven't tested it yet by the way, but is there a way for `hasPathInProject` to accept an wildcard? It'd be much simplear that way e.g. `hasPathInProject("app/layout.*")`